cmd/traffic_cache_tool/CacheTool.cc

```diff
@@ -349,8 +346,10 @@ Stripe::updateHeaderFooter()
     for (int j = 0; j < 2; j++) {
       char *meta_t = (char *)ats_memalign(ats_pagesize(), this->vol_dirlen());
       memcpy(meta_t, &_meta[i][j], sizeof(StripeMeta));
-      CacheStoreBlocks hdr_size = round_up(sizeof(StripeMeta));
-      ssize_t n                 = pwrite(_span->_fd, meta_t, hdr_size, _meta_pos[i][j]);
+      // copy freelist
+      memcpy(meta_t + sizeof(StripeMeta) - sizeof(uint16_t), this->freelist, this->_segments * sizeof(uint16_t));
+      CacheStoreBlocks hdr_size(vol_headerlen());
```

Comment:
I'm concerned the allocation is `vol_dirlen()` while the `pwrite` is `vol_headerlen()`.